### PR TITLE
make receiving events nonblocking even when queue is full

### PR DESCRIPTION
**Pull request by adamdubiel**
Also added two metrics:

* discarded (dropped) events when queue is full
* event queue utilization on 0-100 scale

### CODE REVIEW

File: web/event_handler.go [hunk 1]
```diff
@@ -20,20 +20,22 @@ type event struct {
...
 type eventHandler struct {
-	id              int
-	serviceRegistry service.ServiceRegistry
-	marathon        marathon.Marathoner
-	eventQueue      <-chan event
+	id                 int
+	serviceRegistry    service.ServiceRegistry
+	marathon           marathon.Marathoner
+	eventQueue         <-chan event
+	eventQueueCapacity int
```
Comment (janisz) [inline web/event_handler.go]: Do we really need this value? Can't just use [cap](https://golang.org/pkg/builtin/#cap)

Comment (adamdubiel) [inline web/event_handler.go]: oh great, this is what i was looking for and did not find, will change

**Comment by adamdubiel** [comment]
Are build always this flaky?

**Comment by janisz** [comment]
Tests get flaky when they are run in parallel (multiple packages at once) but in Travis we have run them sequentially. 
Unfortunately your code doesn't pass linter https://travis-ci.org/allegro/marathon-consul/builds/186278642#L217

**Comment by adamdubiel** [comment]
Oh, okay, linter this time. But if you see build history, each time different tests were failing (build continues after 1st fail).

**Comment by adamdubiel** [comment]
Found two places that made tests at my localhost (Mac) fail:

* bumped Consul version to 0.7.2, which fixed issue with Consul segfaults at random moments
* fixed race condition detected by `-race` in Marathon stub

I also added 2 new targets to Makefile:

* `format` that runs `go fmt` on `marathon-consul` sources
* `check` that runs `gometalinter` in same config as on travis

File: web/event_handler.go [hunk 2]
```diff
@@ -55,7 +55,11 @@ func (fh *eventHandler) start() chan<- stopEvent {
 			select {
 			case e = <-fh.eventQueue:
 				metrics.Mark(fmt.Sprintf("events.handler.%d", fh.id))
-				metrics.UpdateGauge("events.queue.len", int64(len(fh.eventQueue)))
+
+				ql := int64(len(fh.eventQueue))
```
Comment (janisz) [inline web/event_handler.go]: Use meaningful name for variable `queueLength`

File: web/event_handler_test.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func TestWebHandler_DropUnknownEventType(t *testing.T) {
 	t.Parallel()
 
 	// given
-	queue := make(chan event)
+	queue := make(chan event, 10)
```
Comment (janisz) [inline web/event_handler_test.go]: Why you use buffered channel?

File: web/event_handler.go (hunk 2 above)
Comment (adamdubiel) [inline web/event_handler.go]: Yeah, looks like the whole go world likes to use those silly one or two letter names, just tried to follow ;) Will change

File: web/event_handler_test.go (hunk 3 above)
Comment (adamdubiel) [inline web/event_handler_test.go]: might not make sense anymore, but it helped to pass any tests at the beginning. will change

**Comment by adamdubiel** [comment]
i'm going to close this pr and try to work on tests. they fail randomly both on localhost and travis when using nonblocking channel.

**Pull request closed without merging**